### DISCUSSION TITLE
Get error "metadata: Metadata must be a python dict object" when importing Companies/Parts/etc through csv/json file

### DISCUSSION THREAD

**Discussion by 2old4it** [inventree/InvenTree#5322] [Q&A]
Way to reproduce:
- Add 1 company in web interface
- do export -> tried with csv and json
- Add 1 company in the export file
- do import in web interface of same export file

After export I get a file with:

`[{"id": 1, "metadata": "{}", "notes": "", "name": "DigiKey", "description": "", "website": "https://www.digikey.com/", "phone": "", "email": "", "contact": "", "link": "", "image": "", "is_customer": "0", "is_supplier": "1", "is_manufacturer": "0", "currency": "USD"}]`

The metadate is filled with two brackets. However if I add another company using the exact same template:

`[{"id": 1, "metadata": "{}", "notes": "", "name": "DigiKey", "description": "", "website": "https://www.digikey.com/", "phone": "", "email": "", "contact": "", "link": "", "image": "", "is_customer": "0", "is_supplier": "1", "is_manufacturer": "0", "currency": "USD"},{"id": 2, "metadata": "{}", "notes": "", "name": "WhatEver", "description": "", "website": "https://www.wahtever.somewhere/", "phone": "", "email": "", "contact": "", "link": "", "image": "", "is_customer": "0", "is_supplier": "1", "is_manufacturer": "0", "currency": "USD"}]`

I get the error: **"metadata: Metadata must be a python dict"** when I try to import the file using the web interface.

My installation is a baremetal one within a python (3.11) virtual environment. It all seams to work.
`git log => commit 93b44ad8e6abdc970969360708aed3ff8189ec93 (HEAD -> stable, tag: 0.12.1, origin/stable)`

Any suggestions?

**Comment by 2old4it** [comment]
Fount it. The json export added quotes `"{}"` in the export. After removing the quotes `{}` it worked.
On the other hand importing a csv file doesn't work. Even if there are no quotes in the csv file the importer added them somewhere.
The result is that:
`./src/InvenTree/InvenTree/models.py:84:        if type(self.metadata) is not dict:`
is **True** as a `print(type(self.metadata))`
will output `<class 'str'>`

**Reply by SchrodingersGat** [reply]
@2old4it do you want to import metadata via this tool? Or would it be simpler to ignore this field?

**Reply by 2old4it** [reply]
@SchrodingersGat I do not want to import metadata. But leaving it at `{}` or removing it completely (as in leaving it empty) will give me the same error (importing it through a csv file). Every csv file that doesn't contain the metadata field works fine.
Leaving the metadata field empty the test in model.py will evaluate to **True** as `type(self.metadata))` translates to `str`.

**Reply by SchrodingersGat** [reply]
So to confirm, removing the "metadata" column from the file you are importing works OK?

**Reply by 2old4it** [reply]
Sorry, no. What I mean is that imports that don't have the metadata column (e.g. Parts or PartsCategory) work fine.

**Reply by SchrodingersGat** [reply]
Ref: https://github.com/inventree/InvenTree/pull/5349
